### PR TITLE
Cmake

### PR DESCRIPTION
**Pull request by jkent**
Moved CMakeLists.txt too standalone/ and created a new CMakeLists.txt that works as an esp-idf component.

### CODE REVIEW

**Comment by chmorgan** [comment]
@phatpaul is it possible to confirm that this works? I'm not actively using libesphttpd.

**Comment by phatpaul** [comment]
I'm not using CMake yet.  At some point when things cool down on my project I may try to switch to CMake.
My understanding is that CMake is optional in esp-idf 3.x and the default in 4.x, and we can choose to keep using makefile in 4.x.  

I can at least check that it doesn't break the existing makefile build...

**Comment by phatpaul** [comment]
@jkent I'll be happy to test it out if you add some short guide on building with CMake to https://github.com/chmorgan/esphttpd-freertos/blob/master/README.md

**Comment by jkent** [comment]
@phatpaul I'll do one better, I'll write a CMakeLists.txt for a proper test... both build systems coexist just fine.  I'll work on that now.  I also noticed my REQUIRES line has espfs in there.  It should be dynamic, so I need to fix that.

**Comment by PaulFreund** [comment]
At the moment I am busy but I will test it with my project as soon as possible!

**Comment by simap** [comment]
@jkent I have it working for me in 4.0-b1, but I had to switch the espfs component to your latest esp32-espfs and back port the relevant api changes to get a cmakeable espfs. 

https://github.com/simap/libesphttpd/commits/cmake_esp32efs

**Comment by jkent** [comment]
Yes, thats expected.  Hopefully we can octopus merge my changes in.

**Comment by qwerty01** [comment]
Any update on merging this PR? This seems to fix the cmake errors with idf.py (at least on Windows).

File: Kconfig [hunk 1]
```diff
@@ -1,22 +1,15 @@
 #ESP-IDF Kconfig configuration file. Not used for ESP8266 RTOS/non-RTOS SDKs.
 menu "esphttpd"
 
-menuconfig ESPHTTPD_ENABLED
```
Comment (phatpaul) [inline Kconfig]: Why did you change this file?  Is it necessary?
It broke some features which were merged that depend on ESPHTTPD_ENABLED

**Comment by chmorgan** [comment]
@phatpaul hey man, have you had a chance to take a look at this pr? Someone is asking for it to resolve a build issue they are seeing.

**Comment by phatpaul** [comment]
Sorry, I have not tested the CMake functionality, since I'm still using GNU Make (legacy) build system for esp32 in my project.  
As for my testing with legacy Make,
I have merged it and found the issue with Kconfig (see my comment in the code above) which breaks my project.  @jkent please revert that file to original or add the missing options back.

I want to get this merged soon to keep up with @jkent development which is diverging.

**Comment by jkent** [comment]
OK, I'll revert ESPHTTPD_ENABLED, I think moving forward, we should remove it because as a component it is redundant.

**Comment by phatpaul** [comment]
Thanks, I don't have a problem with removing the level in menuconfig.  But when you removed it you left a couple of options broken because they still depended on ESPHTTPD_ENABLED.

**Review by phatpaul — APPROVED** [review]
